src/ast/passes/semantic_analyser.cpp

```diff
@@ -3181,7 +3181,11 @@ void SemanticAnalyser::visit(AttachPoint &ap)
                                  << "' does not exist or is not executable";
         break;
       case 1:
-        ap.target = paths.front();
+        // Replace the glob at this stage only if this is *not* a wildcard,
+        // otherwise we rely on the probe matcher. This is not going through
+        // any interfaces that can be properly mocked.
+        if (ap.target.find("*") == std::string::npos)
+          ap.target = paths.front();
```

Comment:
What's the implication of this change? Fixes a broken test on a more sterile system?

Reply:
This will resolve the wildcard ahead of where the mocking takes place. The wildcard is handled perfectly fine later, so I've just made this non-destructive (match using PATH if needed, but don't remove the wildcard that will be expanded later). This matches the semantics below as well, which defers the expansion when there are multiple matches.